Assets/Scripts/Core/UI/InGame/NightShop/CardUI.cs

```diff
@@ -50,50 +65,63 @@ public void Initialize(Card card, GoldController goldController)
         
         public void OnClick()
         {
-            if (isSelected)
+            if (_isSelected)
             {
-                ClientEvents.NightShop ?.OnCardDeselected();
+                ClientEvents.NightShop.OnCardDeselected?.Invoke();
                 return;
             }
-            ClientEvents.NightShop.OnCardSelected(card);
+            ClientEvents.NightShop.OnCardSelected?.Invoke(card);
         }
 
         private void DisableIfNotEnoughGold(int currentGold)
         {
-            GetComponent<Button>().interactable = currentGold >= card.cost;
+            if (currentGold >= card.cost)
+            {
+                button.interactable = true;
+                background.color = _defaultBackgroundColor;
+                innerBackground.color = _defaultInnerBackgroundColor;
+                costText.color = _defaultCostTextColor;
+            }
+            else
+            {
+                button.interactable = false;
+                background.color = disabledBackgroundColor;
+                innerBackground.color = disabledInnerBackgroundColor;
+                costText.color = disabledCostTextColor;
+            }
         }
 
-        void HandleCardSelected(Card card)
+        private void HandleCardSelected(Card selectedCard)
         {
-            if (Equals(this.card, card))
+            if (Equals(card, selectedCard))
             {
                 SelectCard();
                 return;
             }
             DeselectCard();
```

Comment:
Hübscher und verständlicher finde ich:

```suggestion
            if (Equals(card, selectedCard))
                SelectCard();
            else
                DeselectCard();
```